tools/wpt/run.py

```diff
@@ -183,6 +183,20 @@ def install_requirements(self):
     def setup(self, kwargs):
         self.setup_kwargs(kwargs)
 
+def safe_unsetenv(env_var):
+    """Safely remove an environment variable.
+
+    Python3 does not support os.unsetenv in Windows, so we better remove the
+    variable directly from os.environ
+    """
+    try:
+        os.unsetenv(env_var)
+    except AttributeError:
+        try:
+            os.putenv(env_var,"")
```

Comment:
Setting the variable to an empty string isn't the same as deleting it afaik. At least when I experimented on Linux, `del os.environ[KEY]` does actually unset the value. But the other option we have is to move the unsetting into `wptrunner/browsers/firefox.py` so that we delete the variables from the environment we pass down to the Fx binary.

Reply:
unsetenv() is defined for POSIX systems, the putenv("") should only be executed in windows for python < 3.9. Do you prefer us to be more explicit about it (like checking version and platform?)

Reply:
I don't think we need to check version and platform, but what does `del os.environ[env_var]` do on Windows? That works on Linux to unset the environment variable and seems to be the final fallback here. If that doesn't work then something that deletes from os.environ if `unsetenv` exists and otherwise does `os.environ[foo] = ""` seems better than calling `putenv`.

Reply:
Per the linked bug, I think we can just use `del os.environ[key]` everywhere

Reply:
As far as I understand from docs it just deletes the entry from os.environ but does not effectively remove the variable from the environment. I'll double check later with a Windows virtual machine.

Reply:
After checking in a Windows machine with python3.8 it looks like it's better to do the `del os.environ[]` directly as it properly remove the data from the dictionary *and* the environment (I've verified that subprocesses started after calling `del` do not get the environment variable.